ui/browser/views/navbar/location.jsx

```diff
@@ -20,15 +20,52 @@ import { fixURL, getCurrentWebView } from '../../browser-util';
  * to ipcRenderer on mount, and because the key-down event handler is non
  * trivial.
  */
-class Location extends Component {
-  constructor() {
-    super();
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-  }
+const Location = createClass({
```

Comment:
React docs only mention getDefaultState. Need to put those in the class constructor, like this:

class Foo extends React.Component {
  constructor(props) {
    super(props);
    this.state = { ... };
  }
}

Since this is all documented and already a standard among the react croud, let's go with that.